ReactiveObjC/RACSignal.m

```diff
@@ -25,8 +25,56 @@
 #import "RACTuple.h"
 #import <stdatomic.h>
 
+#if DEBUG
+
+#import "RACSourceSymbolExtractor.h"
+
+@interface RACSignal ()
+
+@property (readonly, nonatomic) NSArray<NSString *> *initializationCallStackSymbols;
+
+@property (readonly, nonatomic) NSLock *initializationSourceSymbolLock;
+
+@property (nonatomic) BOOL didExtractInitializationSourceSymbol;
+
+@end
+
+#endif
+
 @implementation RACSignal
 
+#ifdef DEBUG
+
+@synthesize initializationSourceSymbol = _initializationSourceSymbol;
+
+- (instancetype)init {
+  if (self = [super init]) {
+    // Lightweight.
+    _initializationCallStackSymbols = NSThread.callStackSymbols;
+
+    _initializationSourceSymbolLock = [[NSLock alloc] init];
```

Comment:
In this case I'd use either `dispatch_once` on a predicate that is non-static but a field on this instance or an `os_unfair_lock` which is a `uint32_t` under the hood.

Reply:
Regarding the `dispatch_once` - that is what I did in the first place and then came across this documentation ([source](https://developer.apple.com/documentation/dispatch/1447169-dispatch_once)):
```
The predicate must point to a variable stored in global or static scope. The result of using a predicate with automatic or dynamic storage (including Objective-C instance variables) is undefined.
```

Regarding the unfair lock - done

Reply:
Interesting, now I wonder why it has to have a static or global storage.